### PR TITLE
Student - New E2E tests for Assignments

### PR DESCRIPTION
**Pull request by ndrsszsz**
### New E2E tests for Assignments

- testLockedAssignment
- testFutureAssignment
- testExcusedStatus
- testAttemptSelectorAndSubmissionAndRubricScreen

### Updated URLs to fix HelpTests in all 3 apps
- The URL for "Search the Canvas Guides" button in Help screen has changed lately and caused test failures

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



#### Release Note: 
None

#### Affected Apps: None

[MBL-17335](https://instructure.atlassian.net/browse/MBL-17335)

<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/TestsFoundation/TestsFoundation/DataSeeding/Rubric/CreateDSRubricRequest.swift:39:30: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (colon)
    /Users/vagrant/git/TestsFoundation/TestsFoundation/DataSeeding/Rubric/CreateDSRubricRequest.swift:41:97: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (colon)
    /Users/vagrant/git/TestsFoundation/TestsFoundation/DataSeeding/Rubric/CreateDSRubricRequest.swift:71:29: error: Colon Spacing Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (colon)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7fdbc0af55f7c63dd3d7cb6b52d84e9eb6f15187
</p>

**Review by szabinst — APPROVED** [review]

**Review by vargaat — CHANGES_REQUESTED** [review]
Nice work, this is an important area in the app. As I see, we don't test the rubric's description popup, let's add some assertions for that as well

File: Student/StudentE2ETests/Assignments/AssignmentsTests.swift [hunk 1]
```diff
@@ -205,4 +205,196 @@ class AssignmentsTests: E2ETestCase {
...
+
+        rubricRatingOne.hit()
+        let rubricRatingTitle = DetailsHelper.SubmissionComments.rubricRatingTitle(rubric: rubric).waitUntil(.visible)
+        XCTAssertTrue(rubricRatingTitle.isVisible)
+        XCTAssertTrue(rubricRatingTitle.hasLabel(label: rubric.data[0].ratings[0].description))
+
+        rubricRatingZero.hit()
+        XCTAssertTrue(rubricRatingTitle.waitUntil(.visible).isVisible)
+        XCTAssertTrue(rubricRatingTitle.hasLabel(label: rubric.data[0].ratings[1].description))
+    }
```
[the remark this answers is not in the document]
Comment (ndrsszsz) [reply Student/StudentE2ETests/Assignments/AssignmentsTests.swift:407]: Done! 🙂

**Review by szabinst — APPROVED** [review]

**Review by vargaat — APPROVED** [review]

**Pull request merged**